### ISSUE TITLE
Removing boost dependency

### ISSUE THREAD

**Issue by robiwano** [cpp-netlib/cpp-netlib#728]
With C++11, most functionality needed should be supported. Apart from asio of course, but since asio is available in non-boost version, perhaps cpp-netlib could be done in a non-boost version also ? :)

**Comment by glynos** [comment]
Yes, we have discussed this before (even partly implementing it). For version 1.0, removing boost as a dependency will be the plan.

Note that the URI is already independent of Boost.
